### DISCUSSION TITLE
question about the output of  SOAP.create

### DISCUSSION THREAD

**Discussion by Guancred** [SINGROUP/dscribe#142] [Q&A]
Hello,

I want to calculate the soap descriptor to evaluate the disorder of Na atoms in Na3OCl as the shown in the attatched figure (https://pubs.rsc.org/zh-tw/content/articlelanding/2023/ta/d3ta01373a), where the Pcl represent the SOAP descriptor, and i am wondering to know whether the output of SOAP.create function is the Pcl?
![image](https://github.com/SINGROUP/dscribe/assets/91767936/26e514b7-b36f-4ebb-a7f8-46a24f358119)

Thanks very much!
Hong

**Comment by lauri-codes** [comment]
@Guancred: You should probably ask the paper authors more about what they have done there. But the paper does mention that these are "_distributions of the absolute value of the SOAP pCl(r) vector_". I would interpret this sentence as meaning that they calculated the distributions of the lengths for all `SOAP.create` outputs with centers on chlorine atoms.

**Reply by Guancred** [reply]
Thanks, i also calculated the SOAP descriptor for Na3OCl with the center on one chlorine atom, and there are 4185 elements in the feature array. Because i am new for the SOAP descriptor, i do not know what is the meaning of the 4185 elements, and whether average value for the all 4185 elements can character the SOAP for the Cl atom?

Thanks very much for your kind help.

![image](https://github.com/SINGROUP/dscribe/assets/91767936/a2b18212-998b-4ef8-af35-acdd92a80930)

**Reply by lauri-codes** [reply]
If you wish to understand the exact meaning of those 4185 elements, you will need to read more about the theory of the SOAP descriptor itself. But roughly speaking you could think of it as n-dimensional vector pointing to a location in a space of chemical environments. In this space nearby points are chemically similar, while distant ones are dissimilar. This simplification does not paint the full picture but gives you some kind of an idea.

I would say that an average over these values does not make much sense, but _differences_ between two vectors tell you how similar  two environments are. Imagine you have two regular cartesian position vectors `A=[1, 0, 0] `and `B=[0, 0, 1]`. The average values over the elements are the same (1/3), but the difference/distance between these two points is non-zero.

**Reply by Guancred** [reply]
Thanks. 
In addition, you mentioned the "they calculated the distributions of the lengths for all SOAP.create outputs with centers on chlorine atoms", which means this figure is plotted by only the SOAP.create outputs without any post-processing？ such as 

![image](https://github.com/SINGROUP/dscribe/assets/91767936/ff630767-3a4a-4dd9-bcef-1aed0fdfda56)

where the result is different from the reference, maybe i have a wrong understanding. 
![image](https://github.com/SINGROUP/dscribe/assets/91767936/859b9cf3-54df-4025-9407-3700d408c983)
